PhysicsTools/PatAlgos/python/slimming/miniAOD_tools.py

```diff
@@ -101,16 +163,134 @@ def miniAOD_customizeCommon(process):
+                                                   ,'ak8PFJetsCHSSDSubJet:0JecFactor0', 'ak8PFJetsCHSSDSubJet:1JecFactor0'
+                                                   ,'ak8PFJetsCHSSDSubJet:0Flavour'   , 'ak8PFJetsCHSSDSubJet:1Flavour'
+                                                   ,'ak8PFJetsCHSSDSubJet:0CSVv2IVF'  , 'ak8PFJetsCHSSDSubJet:1CSVv2IVF']
 
+    ## Add CMS top tagger subjet info
+    process.ak8PFJetsCHSTopSubJet = cms.EDProducer("RecoJetDeltaRValueMapProducer",
+        src = cms.InputTag("ak8PFJetsCHS"),
+        matched = cms.InputTag("selectedPatJetsCMSTopTagPFJetsCHSPacked"),
+        distMax = cms.double(0.8),
+        values = cms.vstring(
+            "? numberOfDaughters > 0 ? daughterPtr(0).pt : -9999",
+            "? numberOfDaughters > 1 ? daughterPtr(1).pt : -9999",
+            "? numberOfDaughters > 2 ? daughterPtr(2).pt : -9999",
+            "? numberOfDaughters > 3 ? daughterPtr(3).pt : -9999",
+            "? numberOfDaughters > 0 ? daughterPtr(0).eta : -9999",
+            "? numberOfDaughters > 1 ? daughterPtr(1).eta : -9999",
+            "? numberOfDaughters > 2 ? daughterPtr(2).eta : -9999",
+            "? numberOfDaughters > 3 ? daughterPtr(3).eta : -9999",
+            "? numberOfDaughters > 0 ? daughterPtr(0).phi : -9999",
+            "? numberOfDaughters > 1 ? daughterPtr(1).phi : -9999",
+            "? numberOfDaughters > 2 ? daughterPtr(2).phi : -9999",
+            "? numberOfDaughters > 3 ? daughterPtr(3).phi : -9999",
+            "? numberOfDaughters > 0 ? daughterPtr(0).mass : -9999",
+            "? numberOfDaughters > 1 ? daughterPtr(1).mass : -9999",
+            "? numberOfDaughters > 2 ? daughterPtr(2).mass : -9999",
+            "? numberOfDaughters > 3 ? daughterPtr(3).mass : -9999",
+            "? numberOfDaughters > 0 ? daughterPtr(0).jecFactor(0) : -9999",
+            "? numberOfDaughters > 1 ? daughterPtr(1).jecFactor(0) : -9999",
+            "? numberOfDaughters > 2 ? daughterPtr(2).jecFactor(0) : -9999",
+            "? numberOfDaughters > 3 ? daughterPtr(3).jecFactor(0) : -9999",
+            "? numberOfDaughters > 0 ? daughterPtr(0).partonFlavour : -9999",
+            "? numberOfDaughters > 1 ? daughterPtr(1).partonFlavour : -9999",
+            "? numberOfDaughters > 2 ? daughterPtr(2).partonFlavour : -9999",
+            "? numberOfDaughters > 3 ? daughterPtr(3).partonFlavour : -9999",
+            "? numberOfDaughters > 0 ? daughterPtr(0).bDiscriminator('pfCombinedInclusiveSecondaryVertexV2BJetTags') : -9999",
+            "? numberOfDaughters > 1 ? daughterPtr(1).bDiscriminator('pfCombinedInclusiveSecondaryVertexV2BJetTags') : -9999",
+            "? numberOfDaughters > 2 ? daughterPtr(2).bDiscriminator('pfCombinedInclusiveSecondaryVertexV2BJetTags') : -9999",
+            "? numberOfDaughters > 3 ? daughterPtr(3).bDiscriminator('pfCombinedInclusiveSecondaryVertexV2BJetTags') : -9999",
```

Comment:
same here, can you investigate the option of storing daughters as PackedCandidates or some dedicated DF that would avoid 0,1,2,3 and have a proper vector of CandidatePtr occupying zero space when not filled